Week3_제출

besthersoy opened the pull request.

besthersoy committed:
Week3_제출

besthersoy committed:
Week3_제출_v2

Moon-edu commented on tests/homework/q1_create_employee_table.py:
```diff
@@ -22,5 +22,23 @@
 # password: postgres
 """
+import psycopg
+import sys
```
코드 결과에 영향을 주진 않지만 쓰이지 않는 패키지는 깔끔하게 제거하는게 좋습니다.

Moon-edu commented on tests/homework/q6_employee_male.py:
```diff
@@ -16,5 +16,18 @@
+              where gender= 'Male'
+              order by age desc""")
+
+            male= cur.fetchall()
```
```suggestion
            male = cur.fetchall()
```
코딩을 할 때는 일관적인 코딩 스타일도 중요합니다.

Moon-edu commented on tests/homework/q4_insert_visitlog_table.py:
```diff
@@ -43,5 +43,37 @@
+              ,('B00100', '2022-07-11 10:00:00', '2022-07-11 19:00:00', 'work')
+              ,('A08771', '2022-07-11 09:00:00', '2022-07-11 17:56:00', 'work')
+              ,('C00129', '2022-07-11 09:00:00', '2022-07-11 18:20:00', 'work')
+              ,(null    , '2022-07-11 12:30:00', '2022-07-11 14:45:00', 'meeting')
```
python에서는 `null` 대신 `None`

besthersoy committed:
Update tests/homework/q6_employee_male.py

Co-authored-by: Moon-edu <102866148+Moon-edu@users.noreply.github.com>

besthersoy committed:
Week3_제출_v3

null => none,
스페이스 간격 개선

besthersoy committed:
Merge branch 'homework/w3' of https://github.com/besthersoy/de-course-week3 into homework/w3

besthersoy commented:
최종 제출 : Week3_제출v3 입니다.

besthersoy committed:
Week3_제출_v3.2

Moon-edu commented:
실수로 PR을 닫으신 듯하여 제가 다시 PR을 열었습니다. 자동 채점 결과 총 스코어는 75점인데요, 몇 가지 실패 원인에 대해 알려드리고자 합니다.
```
# Test test_create_employees failed
E           psycopg.ProgrammingError: the query has 4 placeholders but 8 parameters were passed

# Test test_cnt_visit_log failed
2023-10-15 13:33:26 WARNING Query 결과 값이 예상 값이 아닙니다. Expected [(7,)], Actual (7,)

Test test_group_by_visit_log_purpose failed
2023-10-15 13:33:27 WARNING Query 결과 값이 예상 값이 아닙니다. Expected [('meeting', 1), ('visit family', 1), ('work', 21)], Actual ('meeting', 1)
```

besthersoy committed:
Week3_제출_v4

#q2. %s 개수 4개 ->8개로 수정
#q7,8. cnt 값이 한개라 fetchone 으로 썼었는데, fetchall로 수정해서 제출해봅니다. q7,8은 어디서 오류가 났는지 잘 모르겠어요..

besthersoy commented:
#q8. group by purpose 의 경우 purpose 별로 cnt 값이 여러개 나올테니 fetchone (x) => fetchall(o) 써야한다고 이해했습니다!

Moon-edu commented:
고생하셨습니다. 모든 테스트가 통과되어 PR을 닫도록 하겠습니다
